src/components/select/simple-select/simple-select.component.js

```diff
@@ -0,0 +1,363 @@
+    };
+  }, [handleGlobalClick]);
+
+  useEffect(() => {
+    return function cleanup() {
+      clearTimeout(filterTimer.current);
+    };
+  }, []);
+
+  function handleTextboxClick(event) {
+    if (onClick) {
+      onClick(event);
+    }
+
+    setOpenState((isAlreadyOpen) => {
+      if (isAlreadyOpen) {
+        return false;
+      }
+
+      if (onOpen) {
+        onOpen();
+      }
+
+      return true;
+    });
+  }
+
+  function handleTextboxFocus(event) {
+    if (onFocus) {
+      onFocus(event);
+    }
+
+    if (openOnFocus && !isFocusBlocked.current) {
+      setOpen(true);
+    }
+
+    isFocusBlocked.current = false;
+  }
+
+  function onSelectOption(optionData) {
+    const { text, value: newValue, selectionType } = optionData;
+    const isClickTriggered = selectionType === 'click';
+
+    updateValue(newValue, text);
+
+    if (selectionType !== 'navigationKey') {
+      setOpen(false);
+      filterText.current = text;
+    }
+
+    if (isClickTriggered) {
+      isFocusBlocked.current = true;
+      textboxRef.focus();
+    }
+  }
+
+  function updateValue(newValue, text) {
+    if (!isControlled.current) {
+      setSelectedValue(newValue);
+      setTextValue(text);
+    }
+
+    if (onChange) {
+      onChange(createCustomEvent(newValue));
+    }
+  }
+
+  function onSelectListClose() {
+    setOpen(false);
+    filterText.current = '';
+    setMatchingText(selectedValue);
+  }
+
+  function isNavigationKey(keyEvent) {
+    return Events.isDownKey(keyEvent) || Events.isUpKey(keyEvent)
+    || Events.isHomeKey(keyEvent) || Events.isEndKey(keyEvent);
+  }
+
+  function preventTextSelection(event) {
+    event.target.selectionStart = event.target.selectionEnd;
+  }
+
+  function getTextboxProps() {
+    return {
+      id,
+      name,
+      inputRef: assignInput,
+      selectedValue,
+      formattedValue: textValue,
+      onClick: handleTextboxClick,
+      onFocus: handleTextboxFocus,
+      onKeyDown: handleTextboxKeydown,
+      onChange: handleTextboxChange,
+      ...textboxProps
+    };
+  }
+
+  return (
+    <StyledSimpleSelect
+      data-component='simple-select'
+      transparent={ transparent }
+      ref={ containerRef }
+    >
+      <SelectTextbox
+        aria-controls={ isOpen ? selectListId : '' }
+        aria-expanded={ isOpen }
+        type='select'
+        onSelect={ preventTextSelection }
+        { ...getTextboxProps() }
+      />
+      { isOpen && (
+        <SelectList
+          ref={ listboxRef }
+          id={ selectListId }
+          anchorElement={ textboxRef.parentElement }
+          onSelect={ onSelectOption }
+          onSelectListClose={ onSelectListClose }
+          highlightedValue={ selectedValue }
+        >
+          { children }
+        </SelectList>
+      ) }
+    </StyledSimpleSelect>
+  );
+};
+
+const valuePropType = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.object
+]);
+
+const formInputPropTypes = {
```

Comment:
It might be worth extracting this into a separate file and then importing it in if it's going to be done for every `Select` variant

Reply:
extracted to SelectTextbox